drivers/usb_c/tcpc/ucpd_numaker.c

```diff
@@ -0,0 +1,2163 @@
+		rc = -EIO;
+		goto cleanup;
+	}
+
+cleanup:
+
+	return rc;
+}
+
+/* Convert VBUS voltage format between H/W bit and mV
+ *
+ * The following factors are taken into consideration:
+ * 1. Analog Vref
+ * 2. UTCPD VBVOL.VBSCALE
+ *
+ * NOTE: UTCPD VBVOL.VBVOL = MSB 10-bit of EADC DAT.RESULT[11:0],
+ *       that is, discarding LSB 2-bit.
+ */
+
+static uint32_t numaker_utcpd_vbus_volt_bit2mv(const struct device *dev, uint32_t bit)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	struct numaker_tcpc_data *data = dev->data;
+
+	__ASSERT_NO_MSG(data->vref_mv);
+	return bit * data->vref_mv * config->utcpd.vbvol.vbscale.value / BIT_MASK(10);
+}
+
+static uint32_t numaker_utcpd_vbus_volt_mv2bit(const struct device *dev, uint32_t mv)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	struct numaker_tcpc_data *data = dev->data;
+
+	__ASSERT_NO_MSG(data->vref_mv);
+	return mv * BIT_MASK(10) / data->vref_mv / config->utcpd.vbvol.vbscale.value;
+}
+
+static int numaker_utcpd_dump_regs(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	UTCPD_T *utcpd_base = config->utcpd_base;
+
+	/* SYS register */
+	NUMAKER_SYS_REG_DUMP(dev, VREFCTL);
+	NUMAKER_SYS_REG_DUMP(dev, UTCPDCTL);
+
+	/* UTCPD register */
+	NUMAKER_UTCPD_REG_DUMP(dev, IS);
+	NUMAKER_UTCPD_REG_DUMP(dev, IE);
+	NUMAKER_UTCPD_REG_DUMP(dev, PWRSTSIE);
+	NUMAKER_UTCPD_REG_DUMP(dev, FUTSTSIE);
+	NUMAKER_UTCPD_REG_DUMP(dev, CTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, PINPL);
+	NUMAKER_UTCPD_REG_DUMP(dev, ROLCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, FUTCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, PWRCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, CCSTS);
+	NUMAKER_UTCPD_REG_DUMP(dev, PWRSTS);
+	NUMAKER_UTCPD_REG_DUMP(dev, FUTSTS);
+	NUMAKER_UTCPD_REG_DUMP(dev, DVCAP1);
+	NUMAKER_UTCPD_REG_DUMP(dev, DVCAP2);
+	NUMAKER_UTCPD_REG_DUMP(dev, MSHEAD);
+	NUMAKER_UTCPD_REG_DUMP(dev, DTRXEVNT);
+	NUMAKER_UTCPD_REG_DUMP(dev, VBVOL);
+	NUMAKER_UTCPD_REG_DUMP(dev, SKVBDCTH);
+	NUMAKER_UTCPD_REG_DUMP(dev, SPDGTH);
+	NUMAKER_UTCPD_REG_DUMP(dev, VBAMH);
+	NUMAKER_UTCPD_REG_DUMP(dev, VBAML);
+	NUMAKER_UTCPD_REG_DUMP(dev, VNDIS);
+	NUMAKER_UTCPD_REG_DUMP(dev, VNDIE);
+	NUMAKER_UTCPD_REG_DUMP(dev, MUXSEL);
+	NUMAKER_UTCPD_REG_DUMP(dev, VCDGCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, ADGTM);
+	NUMAKER_UTCPD_REG_DUMP(dev, VSAFE0V);
+	NUMAKER_UTCPD_REG_DUMP(dev, VSAFE5V);
+	NUMAKER_UTCPD_REG_DUMP(dev, VBOVTH);
+	NUMAKER_UTCPD_REG_DUMP(dev, VCPSVOL);
+	NUMAKER_UTCPD_REG_DUMP(dev, VCUV);
+	NUMAKER_UTCPD_REG_DUMP(dev, PHYCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, FRSRXCTL);
+	NUMAKER_UTCPD_REG_DUMP(dev, VCVOL);
+	NUMAKER_UTCPD_REG_DUMP(dev, CLKINFO);
+
+	return 0;
+}
+
+static int numaker_eadc_vref_init(const struct device *dev)
+{
+	const struct numaker_tcpc_config *const config = dev->config;
+	struct numaker_tcpc_data *data = dev->data;
+	int rc = 0;
+	const struct adc_dt_spec *spec;
+	enum adc_reference reference;
+
+	if (data->vref_mv) {
+		goto cleanup;
```

Comment:
Remove ```goto cleanup``` and just return error value. Same in other functions

Reply:
Removed most `goto cleanup`, keep only needed